### ISSUE TITLE
Update from 0.8.8f to 1.1.37 : SQL error

### ISSUE THREAD

**Issue by belotn** [Cacti/cacti#1547]
When upgrading SQL database from 8.8f to 1.1.37, i ran into multiple sql error for the table automation_devices. Alter table queries fail du to the lack of fields `snmp_version` and `snmp_context`. I update table manually, so it looks like this now : 
```
mysql> describe automation_devices;
+--------------------+-----------------------+------+-----+---------+----------------+
| Field              | Type                  | Null | Key | Default | Extra          |
+--------------------+-----------------------+------+-----+---------+----------------+
| id                 | bigint(20) unsigned   | NO   | PRI | NULL    | auto_increment |
| network_id         | int(10) unsigned      | NO   | MUL | 0       |                |
| hostname           | varchar(100)          | NO   |     |         |                |
| ip                 | varchar(17)           | NO   | UNI |         |                |
| snmp_version       | varchar(255)          | NO   |     | NULL    |                |
| snmp_port          | mediumint(5) unsigned | NO   |     | 161     |                |
| snmp_community     | varchar(100)          | NO   |     |         |                |
| sysName            | varchar(100)          | NO   |     |         |                |
| sysLocation        | varchar(255)          | NO   |     |         |                |
| sysContact         | varchar(255)          | NO   |     |         |                |
| sysDescr           | varchar(255)          | NO   |     |         |                |
| sysUptime          | int(32)               | NO   |     | 0       |                |
| os                 | varchar(64)           | NO   |     |         |                |
| snmp               | tinyint(4)            | NO   |     | 0       |                |
| known              | tinyint(4)            | NO   |     | 0       |                |
| up                 | tinyint(4)            | NO   |     | 0       |                |
| time               | int(11)               | NO   |     | 0       |                |
| snmp_context       | varchar(64)           | YES  |     |         |                |
| snmp_engine_id     | varchar(64)           | YES  |     |         |                |
| snmp_auth_protocol | char(6)               | YES  |     |         |                |
+--------------------+-----------------------+------+-----+---------+----------------+
20 rows in set (0.00 sec)

```
``

**Comment by netniV** [comment]
Do you still have your 0.8.8f version of the database? Would it be possible for you to dump out the structure of it for me to test against?

**Comment by netniV** [comment]
Your table is still quite wrong btw.  I just did a quick comparison with my empty cacti database (which is created using the cacti.sql file in the cacti website folder) and the following differences showed up.

&gt; is mine
&lt; is yours

Any entry without yours means you don't even have that field.

### # differences
```
  +----------------------+-----------------------+------+-----+---------+----------------+
  | Field                | Type                  | Null | Key | Default | Extra          |
  +----------------------+-----------------------+------+-----+---------+----------------+
< | hostname             | varchar(100)          | NO   |     |         |                |
> | hostname             | varchar(100)          | NO   | MUL |         |                |

< | network_id           | int(10) unsigned      | NO   | MUL | 0       |                |
> | network_id           | int(10) unsigned      | NO   |     | 0       |                |

> | snmp_password        | varchar(50)           | YES  |     | NULL    |                |
> | snmp_priv_passphrase | varchar(200)          | YES  |     |         |                |
> | snmp_priv_protocol   | char(6)               | YES  |     |         |                |

> | snmp_username        | varchar(50)           | YES  |     | NULL    |                |

< | snmp_version         | varchar(255)          | NO   |     | NULL    |                |
> | snmp_version         | tinyint(1) unsigned   | NO   |     | 1       |                |
  +----------------------+-----------------------+------+-----+---------+----------------+
```

**Comment by netniV** [comment]
Can you let me know what plugins you have running under 0.8.8 too?

**Comment by netniV** [comment]
Oh and a copy of your entire upgrade database structure if you can please @belotn ?

**Comment by belotn** [comment]
Yes, i have a 8.8f database, i'll send ASAP the structure dump

**Comment by belotn** [comment]
So, active plugins are :
	Aggregate	0.75
	Monitor	1.3		
	Settings	0.71
	Rrdclean	0.41
	Superlinks	1.4
	Thold	0.5
and i join a zip with the two exports

[cacti_sqls.zip](https://github.com/Cacti/cacti/files/1923936/cacti_sqls.zip)

**Comment by netniV** [comment]
Do you or have you ever used Autom8 or Discover plugins?  You seem to have the tables for them in the 0.8.8 structures.

**Comment by belotn** [comment]
Yes,Cycle, Discovery, Boost and Realtime were installed at a time.

**Comment by cigamit** [comment]
What does the audit_database.php script reveal?

**Comment by cigamit** [comment]
Also, did you capture the cacti.log with all the errors?  I guess we would take the sql from the zip file and upgrade it.

**Comment by netniV** [comment]
I don't know about his log file, but I spotted an issue with the upgrader (certainly in the 1_0_0.php file) where if the tables already exist, it is not creating the extra table columns such as snmp_version.  I have not yet checked the other upgrade files to see if they are trying to introduce them later.

**Comment by cigamit** [comment]
Running the audit_database.php after the schema change would at least resolve some of these issues.

**Comment by netniV** [comment]
As I'm running through the 1.2 installer at the moment, I'll add this to my list of things to debug and check.

**Comment by netniV** [comment]
First test failed miserably, we need to check that an upgrade from 0.8.8 can happily work with authentication expectations of 1.2.  It would appear that 1.2 is expecting a permission which is not normally available to a freshly installed admin of 0.8.8

![image](https://user-images.githubusercontent.com/9052188/43040947-7ca2dc2e-8d4a-11e8-9edb-358e2ec96504.png)

**Comment by netniV** [comment]
First test failed miserably, we need to check that an upgrade from 0.8.8 can happily work with authentication expectations of 1.2. It would appear that 1.2 is expecting a permission which is not normally available to a freshly installed admin of 0.8.8

![screen shot 2018-07-21 at 21 28 25](https://user-images.githubusercontent.com/9052188/43040966-beaa2604-8d4a-11e8-9fa6-d54166b92c9b.png)
